ixmp4/data/db/optimization/base.py

```diff
@@ -16,6 +19,9 @@
 
 
 class BaseModel(RootBaseModel, mixins.HasCreationInfo):
+    # NOTE: only subclasses storing data actually define this!
```

Comment:
Defining the type here so that I can access `BaseModel.DataInvalid` in the `utils.validate_data()` function. This might not be necessary though (we could also just raise `OptimizationDataValidationError`s straight from `ixmp4.core.exceptions` there) and it might be confusing since not all subclasses (like `Column`) actually define this error (since they don't all store and validate data). 
So please confirm this solution is fine or please suggest an alternative.